generators/authentication/index.js

```diff
@@ -145,8 +146,8 @@ module.exports = class AuthGenerator extends Generator {
       '@feathersjs/authentication-jwt'
     ];
     const context = Object.assign({
-      kebabEntity: _.kebabCase(this.props.entity),
-      camelEntity: _.camelCase(this.props.entity),
+      kebabEntity: validate(this.props.entity).validForNewPackages ? this.props.entity : _.kebabCase(this.props.entity),
+      camelEntity: validate(this.props.entity).validForNewPackages ? this.props.entity : _.camelCase(this.props.entity),
```

Comment:
I think `camelName` has to stay the same because it is used for variable names and a valid package name is not necessarily a valid variable name.